graphql/resolver.go

```diff
@@ -136,6 +138,29 @@ func New(apiURL string) Config {
 			}
 		}
 
+		if operationContext == SetLastRevisionMutation {
+			projectIdentifier, ok := args["opts"].(map[string]interface{})["projectIdentifier"].(string)
+			if !ok {
+				return nil, InternalServerError.Send(ctx, "finding projectIdentifier for set last revision operation")
+			}
+			project, err := model.FindBranchProjectRef(projectIdentifier)
+			if err != nil {
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("finding project '%s': %s", projectIdentifier, err.Error()))
+			}
+			if project == nil {
+				return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("project '%s' not found", projectIdentifier))
+			}
+			opts := gimlet.PermissionOpts{
+				Resource:      project.Id,
+				ResourceType:  evergreen.ProjectResourceType,
+				Permission:    evergreen.PermissionProjectSettings,
+				RequiredLevel: evergreen.ProjectSettingsEdit.Value,
```

Comment:
Since these opts are shared by `CopyProjectMutation`, `DeleteProjectMutation` and `SetLastRevisionMutation`, I wonder if we can refactor a bit to make it less repetitive. I imagine `CreateProjectMutation` might eventually belong in a different directive like how the legacy UI does it